### PR TITLE
Djinn upgrade Ruby 1.9

### PR DESCRIPTION
**Pull request by madytyoo**

### CODE REVIEW

File: AppController/djinn.rb [hunk 1]
```diff
@@ -2534,10 +2534,11 @@ def restore_appcontroller_state()
     @@secret = json_state['@@secret']
     keyname = json_state['@options']['keyname']
 
+    # Puts json_state.
     json_state.each { |k, v|
       next if k == "@@secret"
       if k == "@nodes"
-        v = Djinn.convert_location_array_to_class(JSON.load(v), keyname)
+         v = Djinn.convert_location_array_to_class(JSON.load(v),keyname)
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Please use two-space indentation for consistency with the rest of the code.

File: AppController/djinn.rb [hunk 2]
```diff
@@ -2578,9 +2579,9 @@ def restore_from_zookeeper()
     json_state = {}
     zookeeper_data['locations'].each { |ip|
       begin
-        Djinn.log_info("Restoring AppController state from ZK at #{ip}")
-        Timeout.timeout(10) do
-          ZKInterface.init_to_ip(HelperFunctions.local_ip(), ip)
+         Djinn.log_info("Restoring AppController state from ZK at #{ip}")
```
Comment (menivaitsi) [inline AppController/djinn.rb]: Please use two-space indentation for consistency with the rest of the code.

File: AppController/lib/cron_helper.rb [hunk 3]
```diff
@@ -32,12 +32,12 @@ def self.update_cron(ip, port, lang, app)
     Djinn.log_debug("saw a cron request with args [#{ip}][#{lang}][#{app}]")
     app_crontab = NO_EMAIL_CRON + "\n"
 
-    if lang == "python27" or lang == "go" or lang == "php"
+    if lang == "python27" or lang  == "go" or lang == "php"
```
Comment (menivaitsi) [inline AppController/lib/cron_helper.rb]: Please remove extra whitespace in 'lang  == "go"'.

File: AppController/lib/datastore_server.rb [hunk 4]
```diff
@@ -69,6 +69,15 @@ def self.start(master_ip, db_local_ip, my_ip, table)
   def self.stop(table)
      MonitInterface.stop(:datastore_server)
   end
+  # The following are needed to comply to the djinn calling in
```
Comment (menivaitsi) [inline AppController/lib/datastore_server.rb]: Why is there a need to distinguish between master and slave at this point? 
Doesn't the method above (self.stop) stop all datastore servers on the machine?

File: AppController/lib/user_app_client.rb [hunk 5]
```diff
@@ -21,12 +21,16 @@ class UserAppClient
   # The port that the UserAppServer binds to, by default.
   SERVER_PORT = 4343
 
+  # This is the minimum Timeout to use when talking to DS.
```
Comment (menivaitsi) [inline AppController/lib/user_app_client.rb]: Does DS refer to the Datastore server(s)?
Please spell it out.

File: AppController/test_client.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+$:.unshift File.join(File.dirname(__FILE__), "lib")
+require 'helperfunctions'
+
+URL = 'https://localhost:17443/'
+
+begin
+  secret = HelperFunctions.get_secret()
+  driver = SOAP::RPC::Driver.new(URL)
+  driver.options["protocol.http.ssl_config.verify_mode"] = nil
+  # Add remote sevice methods
```
Comment (menivaitsi) [inline AppController/test_client.rb]: Add a '.' at the end of the comment.

File: debian/appscale_build.sh [hunk 7]
```diff
@@ -55,17 +41,91 @@ fi
...
+elif [ -n "$(apt-cache search ruby-switch)" ]; then
+        echo "Make sure ruby1.9 is used"
+        apt-get install -y ruby rubygems ruby-switch
+        ruby-switch --set ruby1.9
+fi
+
+# Since the last step in appscale_build.sh is to create the certs directory,
+# its existence indicates that appscale has already been installed.
+if [ -d appscale/.appscale/certs ]; then
+        # Version 2.3.1 and prior didn't have /etc/appscale/VERSION
```
Comment (menivaitsi) [inline debian/appscale_build.sh]: Put a period at the end.

File: debian/appscale_install_functions.sh [hunk 8]
```diff
@@ -354,33 +360,37 @@ postinstallservice()
...
+    fi
 
-  pip_wrapper kazoo
+    pipwrapper kazoo
 }
 
 installpycrypto()
 {
-  pip_wrapper pycrypto
+  pipwrapper pycrypto
```
Comment (menivaitsi) [inline debian/appscale_install_functions.sh]: Make sure indentation matches the rest of this file.

File: AppController/lib/datastore_server.rb (hunk 4 above)
Comment (obino) [inline AppController/lib/datastore_server.rb]: That's correct at this time. If we do that, we may not be able to accept patches for different database though. I would argue that if we want to remove this ability (ie to use different DB) we should do so in a different pull request. This is just to fix an error during shutdown. What do you think?
